packages/grafana-ui/src/components/Combobox/Combobox.tsx

```diff
@@ -412,32 +440,45 @@ export const Combobox = <T extends string | number>(props: ComboboxProps<T>) =>
                 <ul style={{ height: rowVirtualizer.getTotalSize() }} className={styles.menuUlContainer}>
                   {rowVirtualizer.getVirtualItems().map((virtualRow) => {
                     return (
-                      <li
-                        key={`${items[virtualRow.index].value}-${virtualRow.index}`}
-                        data-index={virtualRow.index}
-                        className={cx(
-                          styles.option,
-                          selectedItem && items[virtualRow.index].value === selectedItem.value && styles.optionSelected,
-                          highlightedIndex === virtualRow.index && styles.optionFocused
-                        )}
-                        style={{
-                          height: virtualRow.size,
-                          transform: `translateY(${virtualRow.start}px)`,
-                        }}
-                        {...getItemProps({
-                          item: items[virtualRow.index],
-                          index: virtualRow.index,
-                        })}
-                      >
-                        <div className={styles.optionBody}>
-                          <span className={styles.optionLabel}>
-                            {items[virtualRow.index].label ?? items[virtualRow.index].value}
-                          </span>
-                          {items[virtualRow.index].description && (
-                            <span className={styles.optionDescription}>{items[virtualRow.index].description}</span>
+                      <>
+                        <ComboboxOptionGroup
+                          key={virtualRow.index}
+                          index={virtualRow.index}
+                          item={items[virtualRow.index]}
+                          previousItem={items[virtualRow.index - 1]}
+                          groupClass={styles.optionGroup}
+                          style={{ transform: `translateY(${virtualRow.start}px)` }}
+                        />
```

Comment:
Does this work properly with virtual list height estimation? That's probably why you have the `top: 0` issues.